jdaviz/configs/cubeviz/plugins/slice/slice.py

```diff
@@ -14,7 +14,15 @@
 __all__ = ['Slice']
 
 
-@tray_registry('cubeviz-slice', label="Slice")
+@tray_registry(
+    'cubeviz-slice', label="Slice",
+    viewer_reference_name_kwargs={
+        "_default_spectrum_viewer_reference_name":
+            ["spectrum_viewer_reference_name", {"require_spectrum_viewer": True}],
+        "_default_image_viewer_reference_name":
+            ["image_viewer_reference_name", {"require_image_viewer": True}],
+    }
+)
 class Slice(PluginTemplateMixin):
```

Comment:
I like the idea of what this is trying to accomplish. I'm concerned it's a little opaque to remember exactly "where in the nested tree do I put `require_spectrum_viewer` again??"

Would it be possible to break it out to a separate decorator entirely to something like:

```
@tray_registry('cubeviz-slice', label="Slice")
@viewers_required('spectrum-viewer', 'image-viewer')
class Slice(PluginTemplateMixin):
```

I think something like the above would be easier from a code readability/maintainability standpoint

Reply:
I really like this idea if it's feasible to migrate!  But since this is all internal, we could also postpone this to a future simplification

Reply:
This is similar to my suggestion in https://github.com/spacetelescope/jdaviz/pull/1681#issuecomment-1271627712 (see Points for Discussion #2); I like this a lot too! @duytnguyendtn/@kecnry: let me know if you'd like to see that in this PR or a separate one.

Reply:
I would love to see it in this PR, but I can also understand the desire not let this languish. I'll leave it up to you @bmorris3 if you think it's reasonable to implement now, or if we run the risk with complicated merge conflicts by letting this stale on for long

Reply:
I can try to put it together in the next ~hour, and if it doesn't pan out I'll update you in this thread. 👍🏻